### PR TITLE
storegw: Optimized inject label stage of index lookup.

### PR DESCRIPTION
**Pull request by bwplotka**
```
name                old time/op    new time/op    delta
InjectExtLabels-12    2.16µs ± 4%    0.85µs ± 0%  -60.77%

name                old alloc/op   new alloc/op   delta
InjectExtLabels-12      736B ± 0%      704B ± 0%   -4.35%

name                old allocs/op  new allocs/op  delta
InjectExtLabels-12      2.00 ± 0%      1.00 ± 0%  -50.00%
```

Overall store benchmarks are not great, something is wrong.

```
benchstat -delta-test=none _dev/bench/store_symb_dbefore.txt _dev/bench/store_symb_d.txt
name                                                                old time/op    new time/op    delta
TelemeterRealData_Series/alerts2w/01DN3SK96XDAEKRB1AN30AAW6E-12        841ms ± 0%    1414ms ± 0%  +68.19%
TelemeterRealData_Series/alerts15s/01DN3SK96XDAEKRB1AN30AAW6E-12       284ms ± 0%     352ms ± 0%  +23.89%
TelemeterRealData_Series/subssyncs2w/01DN3SK96XDAEKRB1AN30AAW6E-12     134ms ± 0%      97ms ± 0%  -27.81%
TelemeterRealData_Series/subs2w/01DN3SK96XDAEKRB1AN30AAW6E-12          3.13s ± 0%     2.11s ± 0%  -32.69%
TelemeterRealData_Series/subs15s/01DN3SK96XDAEKRB1AN30AAW6E-12         1.15s ± 0%     0.97s ± 0%  -15.55%

name                                                                old alloc/op   new alloc/op   delta
TelemeterRealData_Series/alerts2w/01DN3SK96XDAEKRB1AN30AAW6E-12        335MB ± 0%     410MB ± 0%  +22.51%
TelemeterRealData_Series/alerts15s/01DN3SK96XDAEKRB1AN30AAW6E-12      94.3MB ± 0%    94.9MB ± 0%   +0.71%
TelemeterRealData_Series/subssyncs2w/01DN3SK96XDAEKRB1AN30AAW6E-12    40.1MB ± 0%    47.8MB ± 0%  +19.27%
TelemeterRealData_Series/subs2w/01DN3SK96XDAEKRB1AN30AAW6E-12          880MB ± 0%    1135MB ± 0%  +29.04%
TelemeterRealData_Series/subs15s/01DN3SK96XDAEKRB1AN30AAW6E-12         228MB ± 0%     228MB ± 0%   +0.00%

name                                                                old allocs/op  new allocs/op  delta
TelemeterRealData_Series/alerts2w/01DN3SK96XDAEKRB1AN30AAW6E-12        1.52M ± 0%     1.52M ± 0%   +0.00%
TelemeterRealData_Series/alerts15s/01DN3SK96XDAEKRB1AN30AAW6E-12       24.0k ± 0%     24.0k ± 0%   -0.00%
TelemeterRealData_Series/subssyncs2w/01DN3SK96XDAEKRB1AN30AAW6E-12      190k ± 0%      190k ± 0%   -0.00%
TelemeterRealData_Series/subs2w/01DN3SK96XDAEKRB1AN30AAW6E-12          3.38M ± 0%     3.38M ± 0%   -0.00%
TelemeterRealData_Series/subs15s/01DN3SK96XDAEKRB1AN30AAW6E-12         19.3k ± 0%     19.3k ± 0%   +0.18%
```

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: pkg/store/bucket.go [hunk 1]
```diff
@@ -743,18 +743,7 @@ func blockSeries(
...
-			if extLset[l.Name] != "" {
-				continue
-			}
-			s.lset = append(s.lset, l)
-		}
-		for ln, lv := range extLset {
-			s.lset = append(s.lset, labels.Label{Name: ln, Value: lv})
-		}
-		sort.Sort(s.lset)
+		s.lset = injectLabels(lset, extLset)
```
Comment (yeya24) [inline pkg/store/bucket.go:746]: Is `lset` already sorted?

**Review by ahmad-1100 — APPROVED** [review]

File: pkg/store/bucket.go [hunk 2]
```diff
@@ -782,6 +771,31 @@ func blockSeries(
 	return newBucketSeriesSet(res), indexr.stats.merge(chunkr.stats), nil
 }
 
+func injectLabels(in labels.Labels, extLset labels.Labels) labels.Labels {
+	out := make(labels.Labels, 0, len(in)+len(extLset))
```
Comment (yeya24) [inline pkg/store/bucket.go:775]: This is already allocated at line 721

File: pkg/store/bucket.go (hunk 1 above)
Comment (bwplotka) [reply pkg/store/bucket.go:746]: yes

File: pkg/store/bucket.go (hunk 2 above)
Comment (bwplotka) [reply pkg/store/bucket.go:775]: oh that's why I see problems lol

Comment (bwplotka) [reply pkg/store/bucket.go:775]: you are golden @yeya24 ! ❤️

**Comment by bwplotka** [comment]
Ok now we are talking!

```
name                                                                old time/op    new time/op    delta
TelemeterRealData_Series/alerts2w/01DN3SK96XDAEKRB1AN30AAW6E-12        841ms ± 0%     667ms ± 0%  -20.74%
TelemeterRealData_Series/alerts15s/01DN3SK96XDAEKRB1AN30AAW6E-12       284ms ± 0%     330ms ± 0%  +16.34%
TelemeterRealData_Series/subssyncs2w/01DN3SK96XDAEKRB1AN30AAW6E-12     134ms ± 0%      73ms ± 0%  -45.29%
TelemeterRealData_Series/subs2w/01DN3SK96XDAEKRB1AN30AAW6E-12          3.13s ± 0%     1.69s ± 0%  -45.83%
TelemeterRealData_Series/subs15s/01DN3SK96XDAEKRB1AN30AAW6E-12         1.15s ± 0%     1.05s ± 0%   -8.79%

name                                                                old alloc/op   new alloc/op   delta
TelemeterRealData_Series/alerts2w/01DN3SK96XDAEKRB1AN30AAW6E-12        335MB ± 0%     313MB ± 0%   -6.48%
TelemeterRealData_Series/alerts15s/01DN3SK96XDAEKRB1AN30AAW6E-12      94.3MB ± 0%    94.1MB ± 0%   -0.21%
TelemeterRealData_Series/subssyncs2w/01DN3SK96XDAEKRB1AN30AAW6E-12    40.1MB ± 0%    37.4MB ± 0%   -6.71%
TelemeterRealData_Series/subs2w/01DN3SK96XDAEKRB1AN30AAW6E-12          880MB ± 0%     828MB ± 0%   -5.86%
TelemeterRealData_Series/subs15s/01DN3SK96XDAEKRB1AN30AAW6E-12         228MB ± 0%     228MB ± 0%   +0.00%

name                                                                old allocs/op  new allocs/op  delta
TelemeterRealData_Series/alerts2w/01DN3SK96XDAEKRB1AN30AAW6E-12        1.52M ± 0%     1.30M ± 0%  -14.69%
TelemeterRealData_Series/alerts15s/01DN3SK96XDAEKRB1AN30AAW6E-12       24.0k ± 0%     21.9k ± 0%   -8.63%
TelemeterRealData_Series/subssyncs2w/01DN3SK96XDAEKRB1AN30AAW6E-12      190k ± 0%      162k ± 0%  -14.68%
TelemeterRealData_Series/subs2w/01DN3SK96XDAEKRB1AN30AAW6E-12          3.38M ± 0%     2.84M ± 0%  -15.89%
TelemeterRealData_Series/subs15s/01DN3SK96XDAEKRB1AN30AAW6E-12         19.3k ± 0%     19.3k ± 0%   +0.08%
```

Thanks @yeya24 for spotting issue!

**Review by yeya24 — APPROVED** [review]
LGTM :tada:

**Review by pracucci — COMMENTED** [review]
Good job! 👏

**Pull request merged**